lib/backend/dynamo/dynamodbbk.go

```diff
@@ -562,6 +572,47 @@ func (b *Backend) Delete(ctx context.Context, key []byte) error {
 	return b.deleteKey(ctx, key)
 }
 
+// ConditionalUpdate updates the matching item in Dynamo if the provided revision matches
+// the revision of the item in Dynamo.
+func (b *Backend) ConditionalUpdate(ctx context.Context, item backend.Item) (*backend.Lease, error) {
+	rev, err := b.create(ctx, item, modeConditionalUpdate)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	item.Revision = rev
+	return backend.NewLease(item), nil
+}
+
+// ConditionalDelete deletes item by key if the provided revision matches
+// the revision of the item in Dynamo.
+func (b *Backend) ConditionalDelete(ctx context.Context, key []byte, rev string) error {
+	if len(key) == 0 {
+		return trace.BadParameter("missing parameter key")
+	}
+
+	av, err := dynamodbattribute.MarshalMap(keyLookup{
+		HashKey:  hashKey,
+		FullPath: prependPrefix(key),
+	})
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	input := dynamodb.DeleteItemInput{Key: av, TableName: aws.String(b.TableName)}
+	input.SetConditionExpression("#v = :rev")
+	input.SetExpressionAttributeNames(map[string]*string{"#v": aws.String("Revision")})
```

Comment:
Doesn't this work? I'm not sure why we do this in `CompareAndSwap`.

```suggestion
	input.SetConditionExpression("Revision = :rev")
```